app/models.py

```diff
@@ -591,57 +591,62 @@ def create(cls, email, name="", password=None, from_partner=False, **kwargs):
 
         return user
 
-    # region Billing
-    def lifetime_or_active_subscription(self) -> bool:
-        """True if user has lifetime licence or active subscription"""
-        if self.lifetime:
-            return True
-
+    def get_active_subscription(
+        self,
+    ) -> Optional[
+        Union[
+            Subscription
+            | AppleSubscription
+            | ManualSubscription
+            | CoinbaseSubscription
+            | PartnerSubscription
+        ]
+    ]:
         sub: Subscription = self.get_subscription()
         if sub:
-            return True
+            return sub
 
         apple_sub: AppleSubscription = AppleSubscription.get_by(user_id=self.id)
         if apple_sub and apple_sub.is_valid():
-            return True
+            return apple_sub
 
         manual_sub: ManualSubscription = ManualSubscription.get_by(user_id=self.id)
         if manual_sub and manual_sub.is_active():
-            return True
+            return manual_sub
 
         coinbase_subscription: CoinbaseSubscription = CoinbaseSubscription.get_by(
             user_id=self.id
         )
         if coinbase_subscription and coinbase_subscription.is_active():
-            return True
+            return coinbase_subscription
 
         partner_sub: PartnerSubscription = PartnerSubscription.find_by_user_id(self.id)
         if partner_sub and partner_sub.is_active():
+            return partner_sub
+
+        return None
+
+    # region Billing
+    def lifetime_or_active_subscription(self) -> bool:
+        """True if user has lifetime licence or active subscription"""
+        if self.lifetime:
             return True
 
-        return False
+        return self.get_active_subscription() is not None
 
     def is_paid(self) -> bool:
         """same as _lifetime_or_active_subscription but not include free manual subscription"""
-        sub: Subscription = self.get_subscription()
-        if sub:
-            return True
-
-        apple_sub: AppleSubscription = AppleSubscription.get_by(user_id=self.id)
-        if apple_sub and apple_sub.is_valid():
-            return True
+        sub = self.get_active_subscription()
+        if sub is None:
+            return False
 
-        manual_sub: ManualSubscription = ManualSubscription.get_by(user_id=self.id)
-        if manual_sub and not manual_sub.is_giveaway and manual_sub.is_active():
-            return True
+        if isinstance(sub, PartnerSubscription):
```

Comment:
ah why do we return False for partner subscription?

Reply:
It's not a paid subscription in SL. Do we want to consider it a paid one? We cannot manage it from SL's side.

Reply:
I think we can consider this subscription as paid in SL too. Btw `User.is_paid` doesn't seem to be used anywhere (rather in the metrics which can be removed) so I think whether a subscription is paid or not doesn't have any effect in SL.

Reply:
Just to confim: It's also used for showing paid referrals. If we want to count them i'll change it.

Reply:
changed to count as paid

Reply:
Yeah it makes sense to count them in referral too.